Validate ID paramters of packngo methods

t0mk opened the pull request:
Fixes #307 This PR adds UUID checking logic (copied from BSD-licensed github.com/google/uuid), and adds checks for all the (programmatically found) id parameters of packngo methods.

If the ID param is an invalid UUID, the methods raise error like:

```
ID "foo" is not 36 char long
```

I still need to fix the unit tests, and optically checked the generated code. Will do tomorrow.

@displague you can take a look on `uuid.go` in the meantime if you want.

displague commented on uuid.go:
```diff
@@ -0,0 +1,56 @@
+// Copyright 2021 Equinix Metal.  All rights reserved.
+// Copyright 2016 Google Inc.  All rights reserved.
```
I think we may want to include the full LICENSE body here:
```
Copyright (c) 2009,2014 Google Inc. All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are
met:

   * Redistributions of source code must retain the above copyright
notice, this list of conditions and the following disclaimer.
   * Redistributions in binary form must reproduce the above
copyright notice, this list of conditions and the following disclaimer
in the documentation and/or other materials provided with the
distribution.
   * Neither the name of Google Inc. nor the names of its
contributors may be used to endorse or promote products derived from
this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```

t0mk commented:
@displague this is ready for review. I fixed the unit tests and added BSD license to `uuid.go`. I looked over the code (which I modified with a python script), and all looks good.
